contributors/devel/strategic-merge-patch.md

```diff
@@ -209,15 +211,115 @@ In an erroneous case, the set may be created with duplicates. Deleting an
 item that has duplicates will delete all matching items.
 
 
-TODO: @pwittrock
-# Changing Patch Format
+# Changing patch format
 
-## Purpose
+As issues and limitations have been discovered with the strategic merge
+patch implementation, it has been necessary to change the patch format
+to support additional semantics - such as merging lists of
+primitives and defining order when merging lists.
 
-## Requirement
+## Requirments for any changes to the patch format
 
-### Version Skew
+**Note:** Changes to the strategic merge patch must be backwards compatible such
+that patch requests valid in previous versions continue to be valid.
 
-## Strategy
+Previously valid patch requests do not need to keep the exact same
+behavior, but do need to behave correctly.
+
+**Example:** if a patch request previously randomized the order of elements
+in a list and we want to provide a deterministic order, we must continue
+to support old patch format but we can make the ordering deterministic
+for the old format.
+
+### Client version skew
+
+Because the server does not publish which patch versions it supports,
+and it silently ignores patch directives that it does not recognize,
+new patches should behave correctly when sent to old servers that
+may not support all of the patch directives.
+
+`kubectl` is a client that must support talking to older and newer server versions,
```

Comment:
I think we should also say that patch from other old clients should work correctly on the new server. This guarantees we are not breaking APIs.